src/app/config/config.model.ts

```diff
@@ -12,12 +12,18 @@ export interface Config {
   octodash: OctoDash;
 }
 
-interface Octoprint {
+export interface Octoprint {
   url: string;
   accessToken: string;
+  basicAuth?: BasicAuth;
   urlSplit?: URLSplit;
 }
 
+export interface BasicAuth {
+  user?: string;
+  pass?: string;
```

Comment:
If those are required in the Schema they shouldn't be optional here either
```suggestion
  user: string;
  pass: string;
```